cvmfs/network/download.cc

```diff
@@ -1688,35 +1624,76 @@ void DownloadManager::FiniHeaders() {
   default_headers_ = NULL;
 }
 
-
-void DownloadManager::Init(const unsigned max_pool_handles,
-                           const perf::StatisticsTemplate &statistics)
+DownloadManager::DownloadManager(const unsigned max_pool_handles,
+                           const perf::StatisticsTemplate &statistics) :
+                  prng_(Prng()),  // added
+                  pool_handles_idle_(new set<CURL *>),
+                  pool_handles_inuse_(new set<CURL *>),
+                  pool_max_handles_(max_pool_handles),
+                  user_agent_(NULL),
```

Comment:
Like `header_list_` and `default_headers_`, this is initialized in `InitHeaders`.
```suggestion
```